File: android/src/main/kotlin/id/my/anggaaryas/flutter_secure_enclave/factory/AccessControlFactory.kt
package id.my.anggaaryas.flutter_secure_enclave.factory

import android.os.Build
import androidx.annotation.RequiresApi
import id.my.anggaaryas.flutter_secure_enclave.model.AccessControlParam

@RequiresApi(Build.VERSION_CODES.M)
class AccessControlFactory(private val value: Map<String, Any?>) {

    fun build(): AccessControlParam {
        return AccessControlParam(value)
    }
}

File: ios/Classes/factory/AccessControlFactory.swift
import Foundation

@available(iOS 11.3, *)
class AccessControlFactory{
    let value: Dictionary<String, Any>
    
    init(value: Dictionary<String, Any>){
        self.value = value
    }
    
    func build() -> AccessControlParam{
        return AccessControlParam(value: value)
    }
}
